### PR TITLE
storage: Polish for the Format dialog

### PR DESCRIPTION
**Pull request by mvollmer**
Release note:

### Storage: the Format dialog has been polished

Fields have been re-ordered and warnings and errors are now consistent with Cockpit's common styling.

image

### CODE REVIEW

**Comment by mvollmer** [comment]
@garrett, I'll work on the storage polish here. Let's try to get it all done for a single release.

**Comment by garrett** [comment]
Our longstanding PatternFly update PR merged, so please also migrate the "alert" warning at the bottom of the dialog to use [helper text](https://www.patternfly.org/v4/components/helper-text) instead.

**Comment by mvollmer** [comment]
About the kebab menu:

image

This looks messy to me, and would be cleaned up if everything would be behind the Kebab menu, including "Create partition".

The reason we have some actions in their own button is to make them more prominent as the "next obvious action".  If "Create partition" would be in a menu, people might struggle to figure out
how to proceed after creating a partition table, for example.

Another example: we make it easy to mount a filesystem by having a prominent "Mount" button, but we make it slightly harder to unmount it by only having "Unmount" in the menu.

**Comment by garrett** [comment]
Good news! There are PatternFly components for these things now.

Quick action buttons should still have a space to the side even if there are no actions.
- composable actions (table variant): https://www.patternfly.org/v4/components/table#composable-actions

Also, if we want to do the indentation, there's a widget for that: 
- tree table (a table variant): https://www.patternfly.org/v4/components/table#tree-table
- composable tree table (another table variant): https://www.patternfly.org/v4/components/table#composable-tree-table
- https://www.patternfly.org/v4/components/tree-view/

In this case, the tree table with composable actions or a composable tree table with actions is what we want.

**Comment by mvollmer** [comment]
Split the content stuff out into #15968

**Comment by mvollmer** [comment]
With HelperText, Alert at top, and the hint that a unmount will happen first:

image

**Comment by garrett** [comment]
> With HelperText, Alert at top, and the hint that a unmount will happen first:

Cool, it's already a lot better!

We should style up the currently in use thing... somehow. (I'm not sure exactly how, but it needs something.) But the rest is already tons better! :+1:

**Comment by mvollmer** [comment]
> We should style up the currently in use thing... somehow.

Yes.  Here is a more complex example:

image

This is for erasing a disk that has multiple partitions, two of which have mounted filesystems, and one is part of a volume group.

**Comment by garrett** [comment]
I wonder if we could use an info helper text for the currently used messages, and ensure there's a bit more spacing between each item (if PF won't add it by default). If using Helper text doesn't add the spacing itself, [additional spacing can be added with PF helper classes](https://www.patternfly.org/v4/utilities/spacing) once they're imported in the SCSS file, such as `pf-u-mt-md` (meaning: PatternFly Utility Margin Top Medium).

**Comment by mvollmer** [comment]
The Format dialog with alignment lines for encryption options.

image

And for the Mount options:

image

**Comment by garrett** [comment]
@KKoukiou pointed out it's the issue I reported @ https://github.com/patternfly/patternfly/issues/3868

**Comment by mvollmer** [comment]
> @KKoukiou pointed out it's the issue I reported @ [patternfly/patternfly#3868](https://github.com/patternfly/patternfly/issues/3868)

Ok, then we don't need to change anything about checkbox alignments here.  Ready for review!

**Comment by mvollmer** [comment]
Oh, wait, the teardown messages.

**Comment by garrett** [comment]
I _think_ I'm on the branch and this is what I'm seeing locally.

Screenshot 2021-06-21 at 14-24-31 Storage - garrett Slate

Screenshot 2021-06-21 at 14-25-38 Storage - garrett Slate

Why is the slider the first thing? Shouldn't it be the name? And then the size should go between the type and erase option?

**Comment by mvollmer** [comment]
> Why is the slider the first thing? Shouldn't it be the name? And then the size should go between the type and erase option?

I'll do that.

**Comment by mvollmer** [comment]
TestStorageFormat.testFormatTypes fails on debina-testing reliably.

**Comment by mvollmer** [comment]
> TestStorageFormat.testFormatTypes fails on debina-testing reliably.

This also fails on master, hmm.

**Comment by mvollmer** [comment]
> > TestStorageFormat.testFormatTypes fails on debina-testing reliably.
> 
> This also fails on master, hmm.

Ahh, this is known issue #1733.

**Review by garrett — APPROVED** [review]
It's definitely getting better:

Screenshot 2021-07-01 at 10-58-37 Storage - garrett Thunder

I'm not a fan of how the "currently in use" block looks. But I guess that would require some design work. Unless you have an idea and want to follow-up with that part of the dialog on this PR, I'll approve for the rest. We can talk about and optimize that later (and should probably open up an issue for that specifically).

File: pkg/storaged/dialog.jsx [hunk 1]
```diff
@@ -318,6 +319,16 @@ function flatten(arr1) {
     return arr1.reduce((acc, val) => Array.isArray(val) ? acc.concat(flatten(val)) : acc.concat(val), []);
 }
 
+const HelperTextWarning = ({ text }) =>
```
Comment (martinpitt) [inline pkg/storaged/dialog.jsx:331]: Any reason why this does not use the [HelperText](https://www.patternfly.org/v4/components/helper-text/) React component? That would also avoid the manual CSS import.

**Review by martinpitt — APPROVED** [review]
Danke! One question, but good enough as-is. Can you please write a little release note and add a current screenshot?

**Review by KKoukiou — APPROVED** [review]
Two comments:
* As far as I know we show dialog errors in the bottom of the dialogs in all other pages
* HelperText component should be used as @martinpitt pointed out

But it's really fine as is, we can post a followup if we agree on the two points above.

File: pkg/storaged/dialog.jsx (hunk 1 above)
Comment (mvollmer) [reply pkg/storaged/dialog.jsx:331]: Oh, I somehow thought it doesn't exist...

**Pull request merged**